docs/plans/2026-07-31-source-backed-diagnostic-renderer.md

```
@@ -0,0 +1,269 @@
+error[E123]: diagnostic message
+--> src/main.mbt
+primary 2:5-2:8: expected expression
+  2 | let x = foo
+    |     ^^^
+secondary 4:1-4:4: declared here
+  4 | foo
+    | ---
+= note: additional context
+```
+
+- The header is lowercase severity, optional `[code]`, `: `, then the diagnostic
```

markdownlint-cli2 (0.23.1)

[warning] Spaces inside code span elements

(MD038, no-space-in-code)